### ISSUE TITLE
make-array is broken

### ISSUE THREAD

**Issue by drmeister** [drmeister/clasp#322]
stassats sez:
```
(make-array 3 :element-type 'bit :initial-contents '(1 1 1)) => #*000
(make-array 3 :element-type '(unsigned-byte 8) :initial-contents '(1 1 1)) => You can only specify one of initial-element or initialContents
```

**Comment by drmeister** [comment]
more problems:
so, core__make_vector the bit case completely ignores any fillings
(make-array 3 :element-type 'character :initial-element #\a) is broken as well
as is (make-array 3 :element-type 'character :initial-element '(#\a #\b #\c))

(make-array 3  :element-type 'character :initial-contents '(#\a #\b #\c)) does work, but :initial-element #\a does not

**Comment by drmeister** [comment]
All fixed.
```
> (make-array 3 :element-type 'bit :initial-contents '(1 1 1))
#*111
> (make-array 3 :element-type '(unsigned-byte 8) :initial-contents '(1 1 1))
#(1 1 1)
> (make-array 3 :element-type 'character :initial-element #\a)
"aaa"
> (make-array 3 :element-type 'character :initial-contents '(#\a #\b #\c))
"abc"
> 
```

**Comment by Shinmera** [comment]
Confirmed fixed on dcc6f10e04e998f628a90922797eeca6dd8b68f9
